pkg/broker/receiver.go

```diff
@@ -214,3 +237,28 @@ func (r *Receiver) shouldSendMessage(ts *eventingv1alpha1.TriggerSpec, event *cl
 	}
 	return true
 }
+
+func addFilteredHeaders(ctx context.Context, tctx cehttp.TransportContext) context.Context {
+	// Helper function that adds the header name and all its values.
+	addHeader := func(ctx context.Context, n string, v []string) context.Context {
+		for _, iv := range v {
+			ctx = cehttp.ContextWithHeader(ctx, n, iv)
+		}
+		return ctx
+	}
+
+	for n, v := range tctx.Header {
+		lower := strings.ToLower(n)
+		if forwardHeaders.Has(lower) {
+			ctx = addHeader(ctx, n, v)
+			continue
+		}
+		for _, prefix := range forwardPrefixes {
+			if strings.HasPrefix(lower, prefix) {
+				ctx = addHeader(ctx, n, v)
+				break
```

Comment:
why do you break here?

Reply:
Seems like this will only add one header?

Reply:
This breaks out of the 'range over forwardPrefixes' for loop, not the 'range over all headers from the context' for loop.